receiver.go

```diff
@@ -45,14 +47,30 @@ func IsForcePush(hook *HookInfo) (bool, error) {
 	return base != hook.OldRev, nil
 }
 
+func (r *Receiver) CheckAllowedBranch(hook *HookInfo) error {
```

Comment:
Made this public so I could add a test. Otherwise, the code in `Handle` requires a full blown setup with commits and repositories and so on. Not ideal, but hope it's alright.